core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -1255,21 +1208,21 @@ private boolean reloadPartitioned() {
 
     private void reloadStruct() {
         // fast path
-        if (this.prevStructVersion == this.structVersion && this.prevPartitionTableVersion == this.partitionTableVersion) {
+        if (this.prevStructVersion == this.txFile.getStructureVersion() && this.prevPartitionTableVersion == this.txFile.getPartitionTableVersion()) {
```

Comment:
we could move `prev` values to txFile

Reply:
I removed most but reader will need prevStructVersion and prevPartitionTableVersion to detect changes in transaction file vs it's open partitions